bin/metrics/memcachedstats.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python
+##############################################################################
+#
+# Copyright (C) Zenoss, Inc. 2019, all rights reserved.
+#
+# This content is made available according to terms specified in
+# License.zenoss under the directory where your Zenoss product is installed.
+#
+##############################################################################
+import argparse
+import logging
+import os
+import re
+import socket
+import sys
+import time
+
+from pprint import pprint
+from gather import ServiceMetrics, MetricGatherer
+
+log = logging.getLogger('zenoss.memcachedmetrics')
+logging.basicConfig()
+log.setLevel(logging.INFO)
+
+
+class MemcachedMetrics(ServiceMetrics):
+    def __init__(self, options):
+        super(MemcachedMetrics, self).__init__(options)
+        self.interval = options.interval
+
+    def build_gatherer(self):
+        return MemcachedMetricGatherer(self.interval)
+
+
+class MemcachedMetricGatherer(MetricGatherer):
+
+    HOST = 'localhost'
+    PORT = 11211
+
+    def __init__(self, interval):
+        self.interval = interval
+        self.prefix = 'zenoss.memcached.'
+        # Get Control Plane ID
+        self.service_id = os.environ.get('CONTROLPLANE_SERVICED_ID', '')
+
+    def _extract_data(self, data):
+        metrics = []
+        tags = {'controlplane_service_id': self.service_id}
+        timestamp = time.time()
+        r = re.compile(r"\bSTAT\b\s(\w*)\s[+-]?([0-9]*[.]?[0-9]+)$")
```

Comment:
Please add a comment here on what this re does. Some of us don't read REs without thought pr are too lazy to ;-)